firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -270,13 +326,49 @@ uint8_t TCMD_parse_full_telecommand(const char tcmd_str[], TCMD_TelecommandChann
         }
     }
 
-    // TODO: read out the @sha256=xx hash and validate it here.
+    // Extract @sha256=xxxx from the telecommand string, starting at &tcmd_str[end_of_args_idx]
+    if (SHA256_ENABLED) {
```

Comment:
Please assume that `SHA256_ENABLED` has been renamed to `TCMD_require_valid_sha256`

This is how I want it to work:

```
if (TCMD_require_valid_sha256 && tag_is_present) -> Validate it. Good.
if (TCMD_require_valid_sha256 && !tag_is_present) -> Error. Good.
if (!TCMD_require_valid_sha256 && !tag_is_present) -> Skip over. Good.
if (!TCMD_require_valid_sha256 && tag_is_present) -> PLEASE CHANGE - perform validation
    Even though the validation is not enabled, please still perform the validation. 
    This will allow a case where we can still protect from corrupted telecommands, but only when the user chooses to send the hash.
```

Reply:
This makes sense, I updated to work under the conditions in your comment.